### DISCUSSION TITLE
Help me with React Native integration

### DISCUSSION THREAD

**Discussion by mdsEu** [react-hook-form/react-hook-form#2029] [Q&A] [answered]
Hi guys, I am needing to know why do I add if the controller component in my react native project this show me the next error: Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but go: undefined, this is the code that I am add:
<Controller
control={control}
render={({ onChange, onBlur, value }) => (
<TextInput
style={styles.input}
onBlur={onBlur}
onChangeText={value => onChange(value)}
value={value}
/>
)}
name="firstName"
rules={{ required: true }}
defaultValue=""
/>
https://react-hook-form.com/get-started#ReactNative
thanks

**Comment by mdsEu** [comment, marked as answer]
I was could resolve this, the error is that documentation use the 'render' word and should to use 'as', please edit this

**Reply by bluebill1049** [reply]
can you do a screenshot where the error is?
